loader_hub/readwise/base.py

```diff
@@ -0,0 +1,54 @@
+"""Simple Reader that loads highlights from readwise.io"""
+import requests
+import json
+from typing import List
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+def _get_readwise_data(api_key, updated_after=None):
+  """
+  Uses Readwise's export API to export all highlights, optionally after a specified date.
+
+  See https://readwise.io/api_deets for details.
+
+  Args:
+      updated_after (datetime.datetime): The datetime to load highlights after. Useful for updating indexes over time.
+  """
+  next_page = None
+  while True:
+    response = requests.get(
+      url="https://readwise.io/api/v2/export/",
+      params={"pageCursor": next_page, "updatedAfter": updated_after},
+      headers={"Authorization": f"Token {api_key}"})
+    response.raise_for_status()
+    yield from response.json()["results"]
```

Comment:
why use `yield from` here if we're not doing streaming? since we're storing all docs in memory anyways with this line `docs = [*_get_readwise_data(...`

Reply:
The function doesn't need to know how it's being used - it could be used for streaming.

I doubt anyone's Readwise DB is big enough to warrant streaming, though. So the reason I used `yield from` is because I think it is tidier.
Since it's a `_private_function()` I figure it doesn't make much difference, but if you'd like me to change it I'd be happy to.